### PR TITLE
Add missing check for global timer for WA

### LINKED ISSUE #590: Waiting_areas - Error control
When using global_timer="true" min_peds is not needed. But if min_peds is not used, the simulation crashes because of the missing value.

### PR DESCRIPTION
**Pull request by schroedtert**
Adds extra check, if global timer for waiting areas is set, if so min_ped do not need to be set.

Fixes #590

### CODE REVIEW

File: libcore/src/IO/GeoFileParser.cpp [hunk 1]
```diff
@@ -837,10 +837,11 @@ Goal * GeoFileParser::parseWaitingAreaNode(TiXmlElement * e)
     // Additional checks:
     const bool waitPed =
         (wa->GetMinNumPed() > 0 && wa->GetMaxNumPed() > 0 && wa->GetWaitingTime() >= 0.);
-    const bool waitDoor = (wa->GetTransitionID() > 0);
+    const bool waitDoor   = (wa->GetTransitionID() > 0);
+    const bool waitGlobal = (wa->IsGlobalTimer() && wa->GetWaitingTime() >= 0.);
 
     // Either (minPed, maxPed, waitingTime) OR transitionID are set
-    if(!waitPed && !waitDoor) {
+    if(!waitPed && !waitDoor && !waitGlobal) {
```
Comment (Ozaq) [inline libcore/src/IO/GeoFileParser.cpp]: The comment above does not fit the code. It should probably just be removed.

File: libcore/src/IO/GeoFileParser.cpp [hunk 2]
```diff
@@ -837,10 +837,11 @@ Goal * GeoFileParser::parseWaitingAreaNode(TiXmlElement * e)
...
         Logging::Error(fmt::format(
             check_fmt("waiting area {:d}: min_peds, max_peds, waiting_time, transition_id not set "
                       "properly. Set either (min_peds, max_peds, waiting_time) or transition_id."),
```
Comment (Ozaq) [inline libcore/src/IO/GeoFileParser.cpp]: The error message needs to be updated to reflect the code change.

File: libcore/src/IO/GeoFileParser.cpp [hunk 3]
```diff
@@ -837,10 +837,11 @@ Goal * GeoFileParser::parseWaitingAreaNode(TiXmlElement * e)
     // Additional checks:
     const bool waitPed =
         (wa->GetMinNumPed() > 0 && wa->GetMaxNumPed() > 0 && wa->GetWaitingTime() >= 0.);
-    const bool waitDoor = (wa->GetTransitionID() > 0);
+    const bool waitDoor   = (wa->GetTransitionID() > 0);
+    const bool waitGlobal = (wa->IsGlobalTimer() && wa->GetWaitingTime() >= 0.);
```
Comment (Ozaq) [inline libcore/src/IO/GeoFileParser.cpp]: You could rename the constants to improve readability quite a bit, e.g.:
```cpp
const bool hasValidConstraints = ...
const bool hasValidTransitionId = ...
const bool usesGlobalTimer = ...
if (!hasValidConstraints && !hasValidTransitionId && !usesGlobalTimer)
{
// handle error
}
```

File: libcore/src/IO/GeoFileParser.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/IO/GeoFileParser.cpp]: comment removed

File: libcore/src/IO/GeoFileParser.cpp (hunk 2 above)
Comment (schroedtert) [reply libcore/src/IO/GeoFileParser.cpp]: done

File: libcore/src/IO/GeoFileParser.cpp (hunk 3 above)
Comment (schroedtert) [reply libcore/src/IO/GeoFileParser.cpp]: done

**Pull request merged**